Lombiq.Tests.UI/Exceptions/CounterThresholdException.cs

```diff
@@ -0,0 +1,64 @@
+using Lombiq.Tests.UI.Services.Counters;
+using System;
+using System.Text;
+
+namespace Lombiq.Tests.UI.Exceptions;
+
+public class CounterThresholdException : Exception
+{
+    public CounterThresholdException()
+    {
+    }
+
+    public CounterThresholdException(string message)
+        : this(probe: null, counter: null, value: null, message)
+    {
+    }
+
+    public CounterThresholdException(string message, Exception innerException)
+        : this(probe: null, counter: null, value: null, message, innerException)
+    {
+    }
+
+    public CounterThresholdException(
+        ICounterProbe probe,
+        ICounterKey counter,
+        ICounterValue value)
+        : this(probe, counter, value, message: null, innerException: null)
+    {
+    }
+
+    public CounterThresholdException(
+        ICounterProbe probe,
+        ICounterKey counter,
+        ICounterValue value,
+        string message)
+        : this(probe, counter, value, message, innerException: null)
+    {
+    }
+
+    public CounterThresholdException(
+        ICounterProbe probe,
+        ICounterKey counter,
+        ICounterValue value,
+        string message,
+        Exception innerException)
+        : base(FormatMessage(probe, counter, value, message), innerException)
+    {
+    }
+
+    private static string FormatMessage(
```

Comment:
Generate exception messages that are easier to understand than:

> DbExecuteCounterKey
> 	SELECT [Document].* FROM [Document] WHERE [Document].[Type] = @Type LIMIT 1
> 		[0]Type = OrchardCore.Settings.SiteSettings, OrchardCore.Settings
> 
> IntegerCounterValue value: 26
> Counter value is greater then DbCommandExecutionRepetitionThreshold, threshold: 22

I as an ordinary developer who didn't set up these counter thresholds but just wanted to change something unrelated but accidentally implemented a `SELECT` N+1 issue, should be able to understand what I did wrong.

Reply:
While the current message is better, it's still cryptic when you first see it.

> SessionProbe, [GET]https://localhost:9341/Admin
Database reader read counter
	Query: SELECT [Document].* FROM [Document] WHERE [Document].[Type] = @Type LIMIT 1
		Parameters: [0]Type = OrchardCore.AdminMenu.Models.AdminMenuList, OrchardCore.AdminMenu
Count: 2
Counter value is greater then SessionThreshold.DbReaderReadThreshold, threshold: 0.

Start the exception with a message that explains, in plain English, what happened.